### PR TITLE
Support for Mastery Precedence rules

### PR DESCRIPTION
**Pull request by Oluwatobiadeoye**
## Summary

The PR represents the studio changes for a new mastery precedence rules that had been recently added to engine(https://github.com/finos/legend-engine/pull/1985). It allows users of mastery to define extra rules around the curation of their master records.

## How did you test this change?

- [x] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
  image
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: .changeset/clean-moose-retire.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@finos/legend-application-studio-deployment': patch
+'@finos/legend-application-studio-bootstrap': patch
+---
+
+Support for Mastery Precedence rules
```
Comment (akphi) [inline .changeset/clean-moose-retire.md:6]: You don't need changeset message for these modules.
I think you missed the only relevant changeset which is "@finos/legend-extension-dsl-mastery"

**Review by akphi — CHANGES_REQUESTED** [review]

File: .changeset/clean-moose-retire.md (hunk 1 above)
Comment (Oluwatobiadeoye) [reply .changeset/clean-moose-retire.md:6]: I had used the command 'yarn changeset' to generate the changeset. Do you still want me to change it?

Comment (akphi) [reply .changeset/clean-moose-retire.md:6]: yeah please change it

Comment (Oluwatobiadeoye) [reply .changeset/clean-moose-retire.md:5]: Thanks. Both comments have now been addresseed.

**Review by akphi — APPROVED** [review]

**Pull request merged**